onnxruntime/core/platform/windows/telemetry.h

```diff
@@ -68,17 +70,17 @@
                                                  ULONGLONG MatchAnyKeyword, ULONGLONG MatchAllKeyword,
                                                  PEVENT_FILTER_DESCRIPTOR FilterData, PVOID CallbackContext)>;
 
-  static void RegisterInternalCallback(const EtwInternalCallback& callback);
+  static void RegisterInternalCallback(const std::string& callback_key, EtwInternalCallback callback);
 
-  static void UnregisterInternalCallback(const EtwInternalCallback& callback);
+  static void UnregisterInternalCallback(const std::string& callback_key);
 
  private:
   static std::mutex mutex_;
   static uint32_t global_register_count_;
   static bool enabled_;
   static uint32_t projection_;
 
-  static std::vector<const EtwInternalCallback*> callbacks_;
+  static std::unordered_map<std::string, EtwInternalCallback> callbacks_;
```

Comment:
>callbacks_ [](http://example.com/codeflow?start=62&length=10)

Because this is a global object, nobody can create any inference session before all such global objects are initialized. Nobody can destroy any session after "main" function is returned.  This is not a well known restriction, and it is a major source of crashes I see.

Reply:
I was investigating another crash.